### ISSUE TITLE
Front-end (inline?) editing

### ISSUE THREAD

**Issue by sarahmonster** [a8cteam51/strikebase#32]
A big important thing we'll need for v2 is front-end editing. @krutal mentioned this in particular way back before we got things started, but this will be important for all to weigh in on, since we'll probably all be entering data. 

Since @michiecat will be doing the majority of initial data entry (I think, see #9) we'll want to make sure we get as much feedback as possible while actually using the app.

While we could just make a custom "edit this project" page, I'd love to see us be able to take advantage of inline editing. My original wireframes (#1) vaguely allude to this. 

From a user perspective, it would be great if you could be logged in and looking at the front-end of the site, notice that let's say Jeeves Butlertron is categorised as a "contractor" when he should be an "influencer", and then you'd just click on the bit of text that says "contractor" and it would magically turn into a drop-down box. You could then change the value of this, click out of the box, and the value would be saved & updated in the DB. 

This would work especially nicely for stuff that often changes, like project status or notes.

I ❤️ inline editing and I think it would be really useful for this in particular, where most of the time we'll be looking up records in order to edit them. It would make working with the app faster, snappier, and way more intuitive. How feasible is this from a technical perspective? How much time would it take? Can we do it? (nudge nudge @danielwrobert)

Some initial thoughts/resources on inline-editing:
http://patternry.com/p=inline-edit/
http://designinginterfaces.com/firstedition/index.php?page=Edit-in-Place
https://webapphuddle.com/inline-edit-design/

**Comment by michiecat** [comment]
I think inline editing would be wonderful to have for a v2, for the reasons you mentioned. It would certainly make using the app faster and more enjoyable (making it more likely for us to continue to use it in the future). 

It'd be good to get an idea of the feasibility of it on wpcom, and what limitations and restrictions on the platform (if any) that we'll face so that we can take it into account ahead of time. 

> Since @michiecat will be doing the majority of initial data entry (I think, see #9) we'll want to make sure we get as much feedback as possible while actually using the app.

Yes, performing the initial data entry will be my primary contribution. :) What sort of feedback would be most useful for you as I enter this data?

**Comment by danielwrobert** [comment]
I think, for a v2, to have something like this we could potentially just render these loops out on the pages using JavaScript and the WP-API. I'm pretty sure everything we'd need should be merged into core now so we don't even need to worry about the plugin. I think sprinkling the features into the traditional theme may be a better approach for a v2, as opposed to re-writing the whole thing (which would take a good bit of time, I feel like). That way we don't need to worry about things like routing and stuff that is handled for us via WordPress. We can circle back to that in the future while still implementing this useful feature now.

With regards to the inline-editing that is described, I would have to look into how much time that would take. I feel like that would be a good bit more involved than having a form somewhere on the page to post from but I could be wrong. Posting from a form to the API is pretty straightforward, though. That said, if the inline-editing is going to be much more difficult/time-consuming, perhaps we can push that to a v3 and add a form to post from (and delete buttons) for a v2?

Either way, I think the inline-editing is definitely worth looking into to better determine how much more involved it would be to implement.

**Comment by michiecat** [comment]
> if the inline-editing is going to be much more difficult/time-consuming, perhaps we can push that to a v3 and add a form to post from (and delete buttons) for a v2?

This sounds like a logical progression to me.

**Comment by sarahmonster** [comment]
> What sort of feedback would be most useful for you as I enter this data?
Basically "what is harder than it needs to be" and "what data are we missing/needs to be better configured." Because Fieldmanager gives us a limited scope for what we can do in terms of data entry, right now it's mostly about getting the data set up in a sensible way. Once we move to v2, we'll be able to fine-tune how that data is actually input.

> as opposed to re-writing the whole thing
👍 Let's not rewrite unless absolutely necessary.

I'm fine with pushing inline editing to a v3, but if it's possible to skip that extra step and just build it out directly, I think that might save us time in the long term. Before we start on v2, let's have a look at how we'd go about implementing it to see a) if the implementation would be completely different from what we'd build for a front-end form and b) how much more work implementing it as inline editing immediately would be.

Once v1 is closed out, we should have something we can start using, so there (should) be less of a rush to get v2 out the door.

**Comment by danielwrobert** [comment]
@sarahmonster I completely agree with you. I have been thinking about this a bit over the weekend. I think, once we have something we can start using in a v1, it may not make much sense to set up multiple version milestones to build out in the future. After all, we are really trying to avoid scope-creep and ending up with a scenario similar to what we discussed with VIP.

Perhaps we should get v1 to something we can start using and then plan for a v2 enhancement. Then in v2, we can maybe just build out the whole front-end with React (and friends) and the WP-API. Here we can add in the front-end editing features. After thinking it over a bit, I believe I have a much better idea of how we can go about building the editing and everything.
